pynicotine/gtkgui/chatrooms.py

```diff
@@ -952,9 +948,14 @@ def server_disconnect(self):
             self.update_room_user_completions()
 
         self.chat_view.update_user_tags()
+        self.chat_entry.set_sensitive(False)
 
     def join_room(self, msg):
         self.populate_room_users(msg.users)
+        self.chat_entry.set_sensitive(True)
+
+    def leave_room(self, _msg):
+        self.server_disconnect()
```

Comment:
Put the code from `server_disconnect()` in here, and then call this function from there instead.